truffle/src/main/java/org/jruby/truffle/nodes/core/BigDecimalNodes.java

```diff
@@ -676,36 +677,51 @@ public Object mult(RubyBasicObject a, RubyBasicObject b, int precision) {
 
     }
 
-    @CoreMethod(names = "/", required = 1)
+    @CoreMethod(names = {"/", "quo"}, required = 1)
     public abstract static class DivOpNode extends BigDecimalCoreMethodNode {
 
         public DivOpNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
         }
 
+        final ConditionProfile normalZero = ConditionProfile.createBinaryProfile();
+
         @TruffleBoundary
-        private Object divBigDecimal(RubyBasicObject a, BigDecimal b) {
-            // precision based on https://github.com/pitr-ch/jruby/blob/bigdecimal/core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java#L892-903
-            final int len = getBigDecimalValue(a).precision() + b.precision();
-            final int pow = len / 4;
-            final int precision = (pow + 1) * 4 * 2;
+        private Object div(RubyBasicObject a, BigDecimal b) {
+            if (normalZero.profile(b.signum() == 0)) {
+                switch (getBigDecimalValue(a).signum()) {
+                    case 1:
+                        return createRubyBigDecimal(Type.POSITIVE_INFINITY);
+                    case 0:
+                        return createRubyBigDecimal(Type.NAN);
+                    case -1:
+                        return createRubyBigDecimal(Type.NEGATIVE_INFINITY);
+                    default:
+                        throw new RuntimeException();
+                }
+            } else {
+                // precision based on https://github.com/pitr-ch/jruby/blob/bigdecimal/core/src/main/java/org/jruby/ext/bigdecimal/RubyBigDecimal.java#L892-903
```

Comment:
Not sure think this kind of comment is a good idea - the link and line numbers will bit-rot very quickly.

Could you maybe format as:

```java
// Precision based on RubyBigDecimal#op_quo19_20 - see copyright and authorship there
```